python/venv/private/venv.bzl

```diff
@@ -214,11 +210,65 @@ def _py_venv_binary_impl(ctx):
             env_inherit = [],
             targets = ctx.attr.data,
         ),
-        OutputGroupInfo(
-            python_zip_file = depset([python_zip_file]),
+        PyMainInfo(
+            main = ctx.file.main,
+            srcs = ctx.files.srcs,
         ),
     ]
 
+def _py_venv_zipapp_impl(ctx):
+    venv_toolchain = ctx.toolchains[venv_common.TOOLCHAIN_TYPE]
+    py_info = ctx.attr.binary[PyInfo]
+    main_info = ctx.attr.binary[PyMainInfo]
+
+    python_zip_file = venv_common._create_python_zip_file(
```

Comment:
Why not remove this from `venv_common`?

Reply:
I can export it separately if you prefer.  it uses helpers in venv_common.bzl